cmd/octopus/main.go

```diff
@@ -22,15 +23,24 @@ func main() {
 	// if there is a missing or invalid .env file anywhere, we don't care, just ignore it
 	_ = godotenv.Load()
 
-	clientFactory, err := apiclient.NewClientFactoryFromEnvironment()
+	// initialize our wrapper around survey, which is also used as a flag for whether
+	// we are in interactive mode or automation mode
+	askProvider := question.NewAskProvider(survey.AskOne)
```

Comment:
@domenicsim1 this is significant. Both the ClientFactory and the proper factory need a reference to Survey, and it gets nilled out when you specify --no-prompt on the command line. I had to make a third struct to hold the Survey reference to avoid an import loop between ClientFactory <-> Factory.

Seems to work well enough